src/Scene/Scene.js

```diff
@@ -131,27 +111,65 @@ Scene.prototype.scheduleUpdate = function scheduleUpdate(forceRedraw) {
 
     if (this.renderingState !== RENDERING_ACTIVE) {
         this.renderingState = RENDERING_ACTIVE;
+        if (__DEBUG__) {
+            document.title += ' ⌛';
+        }
 
         requestAnimationFrame(() => { this.step(); });
     }
 };
 
-Scene.prototype.update = function update() {
-    for (var l = 0; l < this.layers.length; l++) {
-        var layer = this.layers[l].node;
-
-        for (var sl = 0; sl < layer.children.length; sl++) {
-            var sLayer = layer.children[sl];
-
-            if (sLayer instanceof Quadtree) {
-                this.browserScene.updateQuadtree(this.layers[l], this.map.layersConfiguration, this.currentCamera());
-            } else if (sLayer instanceof MobileMappingLayer) {
-                this.browserScene.updateMobileMappingLayer(sLayer, this.currentCamera());
-            } else if (sLayer instanceof Layer) {
-                this.browserScene.updateLayer(sLayer, this.currentCamera());
+Scene.prototype.addGeometryLayer = function addGeometryLayer(layer) {
+    if (typeof (layer.update) !== 'function') {
+        throw new Error('Cant add GeometryLayer: missing a update function');
+    }
+    if (typeof (layer.preUpdate) !== 'function') {
+        throw new Error('Cant add GeometryLayer: missing a preUpdate function');
+    }
+    this._geometryLayers.push(layer);
+};
+
+Scene.prototype.getAttachedLayers = function getAttachedLayers(filter) {
+    const result = [];
+    for (const geometryLayer of this._geometryLayers) {
+        for (const attached of geometryLayer._attachedLayers) {
+            if (!filter || filter(attached, geometryLayer)) {
+                result.push(attached);
             }
         }
     }
+    return result;
+};
+
+function updateElement(context, geometryLayer, element) {
```

Comment:
```js
function updateElements(context, geometryLayer, elements) {
    if (!elements) {
        return;
    }
    for (const element of elements) {
        // update attached layers
        for (const attachedLayer of geometryLayer._attachedLayers) {
            attachedLayer.update(context, attachedLayer, element);
        }
        // update element
        const newElementsToUpdate = geometryLayer.update(context, geometryLayer, element);
        updateElements(context, geometryLayer, newElementsToUpdate);
    }
}

Scene.prototype.update = function update() {
    const context = {
        camera: this.gfxEngine.camera,
        scheduler: this.scheduler,
        scene: this,
    };

    for (const geometryLayer of this._geometryLayers) {
        const elementsToUpdate = geometryLayer.preUpdate(context, geometryLayer);
        updateElements(context, geometryLayer, elementsToUpdate);
    }
};
```
otherwise, `updateElements` is recursive function, elements are then necessarily hierarchical.
Therefore it would be clear that those are children.
So replace `Elements` by ` Children` or `ChildElement`?